### PR TITLE
ENH: linalg: update `_procrustes.py` to handle complex matrices (#1)

### PR DESCRIPTION
**Pull request by keithbriggs**
* Update _procrustes.py

### CODE REVIEW

**Comment by melissawm** [comment]
I believe adding a test for the complex case would be appropriate.

**Comment by ilayn** [comment]
Also needs some PEP8 line length cleanup

**Comment by mdhaber** [comment]
If tests pass, I'll go ahead and merge. Feel free to review, but seems straightforward enough assuming there are no caveats to the math (which seems to check out).

File: scipy/linalg/_procrustes.py [hunk 1]
```diff
@@ -72,6 +73,24 @@ def orthogonal_procrustes(A, B, check_finite=True):
...
+
+    >>> shape = (4, 4)
+    >>> rng = np.random.default_rng(589234981235)
+    >>> A = rng.random(shape) + rng.random(shape)*1j
+    >>> Q = rng.random(shape) + rng.random(shape)*1j
+    >>> Q, _ = np.linalg.qr(Q)
+    >>> B = A @ Q
+
+    `orthogonal_procrustes` recovers the unitary matrix ``Q``
+    from ``A` and ``B``.
```
Comment (mdhaber) [inline scipy/linalg/_procrustes.py:88]:
```suggestion
    from ``A`` and ``B``.
```

**Comment by keithbriggs** [comment]
Thanks for merging!

**Pull request merged**